modules/local/plot_run_gantt/main.nf

```diff
@@ -2,7 +2,7 @@ process PLOT_RUN_GANTT {
     tag "$meta.id"
 
     conda 'click=8.0.1 pandas=1.1.5 plotly_express=0.4.1 typing=3.10.0.0'
-    container 'seqeralabs/nf-aggregate:click-8.0.1_pandas-1.1.5_plotly_express-0.4.1_typing-3.10.0.0--ccea219dc6c3d6a1'
+    container 'community.wave.seqera.io/library/click_pandas_plotly_express_typing:21adb9e2d1b605a5'
```

Comment:
NOOOOOOOOOO

Reply:
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdHJzM3lrcGlqajh6ZzhxbjZ1N3dxbHJocG1ieGFxbjU3dDh0YXNkYSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/kQOxxwjjuTB7O/giphy.gif)

Reply:
Have we figured out how to run these pipelines in offline environments? Is this going to be inherently taken care of by `nf-core download` and not handled by NF anymore @ewels ?

Reply:
No - if we want to run offline then we cannot use Wave.

The only way that I can see around this is to have the Nextflow Wave plugin create the conda + dockerfile locally and from that figure out what the image name should be. Then this could be checked against the local docker / singularity cache without doing any calls to the internet. But I've spoken to @pditommaso about this already and I think that it was difficult / impossible.

So until we figure that out, I can't see any way to use Wave directly in nf-core pipelines, as it will break all offline usage. So the hardcoded docker + singularity URIs have to stay.

We can of course use Wave to build those images and so my hope is that the developer can specify just the conda requirements and then the automation will pin the image names into a config without human intervention.

If anyone has any better ideas, shout!

Reply:
We are near to have `nextflow inspect` to report the HTTP uri of the Singularity image files. Would not that allow handling them in the same manner as it's done for Biocontainers right now?

Reply:
> Is it possible to use the hash and check for existing containers before hitting Wave?

Wave generates the hash, so no